### ISSUE TITLE
the injector not working

### ISSUE THREAD

**Issue by MikaCybertron** [reveny/Android-LD-Preload-Injector#1]
the injector not work because i have tried it to inject it before open the game and even tried to inject it while in-game but still not working

can you make a video preview how to make it work??

**Comment by WhiskyModz** [comment]
@MikaCybertron @reveny What is the difference between the previous injector and this injector?

**Comment by MikaCybertron** [comment]
> @MikaCybertron @reveny What is the difference between the previous injector and this injector? so is pre_load more useful?

this one don't use ptrace to inject the lib, but until now i can not make it work, so i hope @reveny can give a hint or video preview about how to make it work

**Comment by reveny** [comment]
There are 4 things that could go wrong here.
1. The library you want to Inject does not match the architecture of the device
2. The library does not have the correct permissions: chmod a+rx (this has been fixed in the newest release)
3. SELinux is not set to Permissive (command: setenforce 0)
4. Download the newest release apk

How to Inject:
1. Select a package name 
2. Select a library path (needs to be at /data/local/tmp/)
3. Click Inject
5. Close the Injector app and launch the app/game manually (In my case the game takes a bit longer to load)
6. The app/game should have loaded your library.
7. Please do not rely on ImGui to check if the Injection works, check the logcat.

Game and Device tested on:
Device: Samsung Galaxy S20 (android 11)
Game: Rider (com.ketchapp.rider) https://rider.en.uptodown.com/android
Library: https://www.mediafire.com/file/5rjslg63r2ty3q8/libs.zip/file (log tag is "RevenyTest")

**Comment by MikaCybertron** [comment]
> There are 4 things that could go wrong here.
> 
> 1. The library you want to Inject does not match the architecture of the device
> 2. The library does not have the correct permissions: chmod a+rx (this has been fixed in the newest release)
> 3. SELinux is not set to Permissive (command: setenforce 0)
> 4. Download the newest release apk
> 
> How to Inject:
> 
> 1. Select a package name
> 2. Select a library path (needs to be at /data/local/tmp/)
> 3. Click Inject
> 4. Close the Injector app and launch the app/game manually (In my case the game takes a bit longer to load)
> 5. The app/game should have loaded your library.
> 6. Please do not rely on ImGui to check if the Injection works, check the logcat.
> 
> Game and Device tested on: Device: Samsung Galaxy S20 (android 11) Game: Rider (com.ketchapp.rider) https://rider.en.uptodown.com/android Library: https://www.mediafire.com/file/5rjslg63r2ty3q8/libs.zip/file (log tag is "RevenyTest")

ahh so this means that the library must match with OS Architectures, for example like the library i'm gonna use for inject is armv7 and then inject it to emulator, then it won't work because the OS Architecture is x86  even though the emulator supports armv7 and arm64.

so this means the way still same like injector with ptrace, because the library architectures must match with architectures of the device (OS Architectures)

i'm correct??

**Comment by reveny** [comment]
Correct.

**Comment by rev** [comment]
First of all, hello fellow Rev! Secondly, I think you would benefit from something like this: https://github.com/Rev/Architecture-Checker

**Comment by reveny** [comment]
> First of all, hello fellow Rev! Secondly, I think you would benefit from something like this: https://github.com/Rev/Architecture-Checker

Thank you for your reply, for LDPreload Injection this is not really necessary as it turns out that LDPreload does not work on most game emulators such as ldplayer. I will however use the project for my DLL-Injector project. Your way of determining the apps architecture is pretty good and I have not thought of doing it that way myself, thanks for sharing 👍

**Comment by ghost** [comment]
POCO X3 Pro with xiaomi.eu custom rom (MIUI) doesn't work. @reveny your phone has a custom rom?

**Comment by reveny** [comment]
> 

No, my phone does not have a custom rom but like I mentioned before, some phones do not support LD-Preload for some reason

**Comment by xyxdaily** [comment]
I had tried in pixel 3axl android12, however it does not work. And I tried to use the cmd to input the command.
setenforce 0 
setprop wrap.mypackname LD_PRELOAD=/data/local/tmp/libtest.so
setprop mypackname LD_PRELOAD=/data/local/tmp/libtest.so

Then start mypackname , howerver the libtest.so is not worked.

If I follow this url(https://www.cnblogs.com/sandeepin/p/ld-preload-inject.html), the LD_Preload can work.

**Comment by huaerxiela** [comment]
mokee android10, the ld_preload injector type working success, but see log can find the inject so execute many times,
I wonder if it has something to do with the number of so in apk?

**Comment by brandonros** [comment]
```
08-08 23:37:55.038 16623 16623 F linker  : CANNOT LINK EXECUTABLE "/system/bin/app_process32": couldn't map "/data/local/tmp/libPreload-armeabi-v7a.so" segment 1: Permission denied
```

```
1|gta7litewifi:/ # ls -lh /data/local/tmp/libPreload-armeabi-v7a.so                                                                                                                                                                                                                                                      
-rwxrwxrwx 1 shell shell 5.0K 2023-08-08 23:35 /data/local/tmp/libPreload-armeabi-v7a.so
```

Weird...
